src/backend/tty.rs

```diff
@@ -970,11 +970,14 @@ impl Tty {
 
         niri.add_output(output.clone(), Some(refresh_interval(mode)), vrr_enabled);
 
-        // Power on all monitors if necessary and queue a redraw on the new one.
-        niri.event_loop.insert_idle(move |state| {
-            state.niri.activate_monitors(&mut state.backend);
-            state.niri.queue_redraw(&output);
-        });
+        // redraw is always performed when when monitors are reactivated
+        if niri.monitors_active {
+            // Power on all monitors if necessary and queue a redraw on the new one.
+            niri.event_loop.insert_idle(move |state| {
+                state.niri.activate_monitors(&mut state.backend);
+                state.niri.queue_redraw(&output);
+            });
+        }
```

Comment:
Something like this I guess
```suggestion
        } else {
                set_crtc_active(&device.drm, *crtc, false);
                if let Err(err) = surface.compositor.reset_state() {
                    warn!("error resetting surface state: {err:?}");
                }
        }
```